src/containers/settings/user-profile.tsx

```diff
@@ -46,10 +52,14 @@ class UserProfile extends React.Component<RouteComponentProps, IState> {
         this.initialState = { ...result };
         this.setState({ user: result });
       })
-      .catch(() => this.props.history.push(Paths.notFound));
+      .catch(() => this.setState({ redirect: Paths.notFound }));
   }
 
   render() {
+    if (this.state.redirect) {
+      return <Redirect to={this.state.redirect} />;
```

Comment:
:+1:, but also https://github.com/ansible/ansible-hub-ui/pull/859#issuecomment-910920191

```suggestion
      return <Redirect push to={this.state.redirect} />;
```

Reply:
Ok.